chapters/02.Quadratic/quadratic.jl

```diff
@@ -0,0 +1,1968 @@
+	"""
+	function quadratic_ls(P::AbstractMatrix, q::AbstractVector, Δx::AbstractVector,
+										x::AbstractVector)
+		t = solve_quadratic(Δx' * P * Δx, Δx' * P * x + Δx ⋅ q)
+		return t
+	end
+
+	"""
+		gradient_descent(P::AbstractArray, q::AbstractVector,
+				x₀::AbstractVector; β::Real=0.0, ϵ::Real=1e-6,
+				tracker::Tracker=notrack)
+
+	Computes the minimizes of a quadratic system using gradient descent. Optionally
+	provide momentum.
+
+	Inputs:
+		- P, q: the terms of the nD quadratic system
+		- x₀: starting point
+		- ϵ: convergence parameter
+		- β: momentum parameter
+		- track: bool to indicate whether to store the path
+
+	Outputs:
+		- xstar: the found minimum
+	"""
+	function gradient_descent(P::AbstractArray, q::AbstractVector,
+				x₀::AbstractVector; β::Real=0.0, ϵ::Real=1e-6,
+				track=false)
+		@assert 0 ≤ β < 1
+		x = x₀  # initial value
+		Δx = zero(x)  # pre-allocate a vector for the gradient
+		track && (path = [copy(x)])
+		while true
+			Δx .= (1.0 - β) * (- P * x .- q) .+ β *  Δx
+			if norm(Δx) < ϵ
+				break
+			end
+			# determine stepsize using exact line search
+			t = quadratic_ls(P, q, Δx, x)
+			# perform step
+			x .+= t * Δx
+			track && push!(path, copy(x))
+		end
+		if track
+			return x, path
+		else
+			return x
+	end
+	end
+
+
+end
+
+# ╔═╡ 752ced8e-1b80-11ec-298c-050f319f7228
+md"""
+# Quadratic optimization
+
+*STMO*
+
+**Michiel Stock**
+
+![](https://github.com/MichielStock/STMO/blob/master/chapters/02.Quadratic/Figures/logo.png?raw=true)
+"""
+
+# ╔═╡ c7c39058-1035-4b34-8209-cc71837b6b38
+md"""
+## Motivation
+
+Quadratic systems are essential! Firstly, we can often closely approximate systems close to their maximum by a quadratic system. Studying the minimization of quadratic systems can teach us about the minimization of general convex functions. Quadratic systems are also crucial in their own right! Many statistical models, graph problems, molecular models, etc. can be formulated as quadratic systems:
+
+- least-square minimization problems
+- inference using multivariate normal distributions
+- molecular modeling using spring-mass systems
+- signal recovery
+
+"""
+
+# ╔═╡ 0a65fdd9-5579-4a60-a826-d438b6a49caf
+md"""
+## Warming up: one-dimensional quadratic systems
+
+In the scalar case, a quadratic function is given by
+
+$$f(x) = \frac{1}{2}px^2+qx +r\,,$$
+
+with $p>0$ (we will shortly see why).
+
+Our optimization problem is given by:
+
+$$\min_x\,\frac{1}{2}px^2+qx +r\,.$$
+
+This can easily be solved by setting the first order derivative equal to zero:
+
+$$\frac{\mathrm{d}f(x)}{\mathrm{d}x} = px + q$$
+
+$$px^\star+q = 0 \Leftrightarrow x^\star=\frac{-q}{p}\,.$$
+
+To show that this is the sole minimizer of $f(x)$, we have to prove that the second-order derivative is positive at this point. This means that at that point, the derivative of the function is increasing: a little to the left the function is increasing, a little to the right and the function is decreasing. We have
```

Comment:
This means that at that point, the derivative of the function is increasing: a little to the left the function is **decreasing**, a little to the right and the function is **increasing**?